frontend/src/Components/UserProfile.js

```diff
@@ -1,13 +1,79 @@
-import React, { useContext } from "react";
+import React, { useEffect, useState, useContext } from "react";
 import { AuthContext } from "../Context/AuthContext";
+import Post from "./Post/Post";
+import axios from "axios";
+import { Loading } from "../Components/Common/Loader";
+import { POST_GET_ALL_URL } from "../constants";
 
-export default function UserFeed() {
+export default function UserProfile() {
   const { user, token } = useContext(AuthContext);
+  const [posts, setPosts] = useState([]);
+  const [loading, setLoading] = useState(false);
+
+  useEffect(() => {
+    setLoading(true);
+    axios
+      .get(POST_GET_ALL_URL, {
+        headers: {
```

Comment:
You're using the wrong Axios URL. You need to either use `/user/current_user` or (`/user/details` with user_id).

This is calling the _whole_ Posts URL.